core/src/main/scala/org/apache/spark/scheduler/BlacklistStrategy.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import scala.collection.mutable
+import org.apache.spark.Logging
+import org.apache.spark.SparkConf
+import org.apache.spark.util.SystemClock
+import org.apache.spark.util.Clock
+
+/**
+ * The interface to determine executor blacklist and node blacklist.
+ */
+private [scheduler] trait BlacklistStrategy {
+  /** Defined a time interval to expire failure information of executors */
+  val expireTimeInMilliseconds: Long
+
+  /** Return executors in blacklist which are related to given taskIndex */
+  def getExecutorBlacklist(
+      executorIdToFailureStatus: mutable.HashMap[String, FailureStatus],
+      atomTask: BlacklistAtomTask, clock: Clock): Set[String]
+
+  /** Return all nodes in blacklist */
+  def getNodeBlacklist(
+      executorIdToFailureStatus: mutable.HashMap[String, FailureStatus]): Set[String]
+
+  /**
+   * Default implementation to remove failure executors from HashMap based on given time period.
+   * The return value identity whether or not it updated anything
+   */
+  def expireExecutorsInBlackList(
+      executorIdToFailureStatus: mutable.HashMap[String, FailureStatus], clock: Clock): Boolean = {
+    val now = clock.getTimeMillis()
+    val expiredKey = executorIdToFailureStatus.filter {
+      case (executorid, failureStatus) => {
+        (now - failureStatus.updatedTime) >= expireTimeInMilliseconds
+      }
+    }.keySet
+
+    if (expiredKey.isEmpty) {
+      false
+    } else {
+      executorIdToFailureStatus --= expiredKey
+      true
+    }
+  }
+}
+
+/**
+ * This strategy is simply based on given threshold and is taskId unrelated. An executor will be
+ * in blacklist, if it failed more than "maxFailureTaskNumber" times. A node will be in blacklist,
+ * if there are more than "maxBlackExecutorNumber" executors on it in executor blacklist.
+ *
+ * In this case, provided taskId will be ignored. The benefit for taskId unrelated strategy is that
+ * different taskSets can learn experience from other taskSet to avoid allocating tasks on
+ * problematic executors.
+ */
+private[scheduler] class SimpleStrategy(
+    maxFailureTaskNumber: Int,
+    maxBlackExecutorNumber: Int,
```

Comment:
can you change these params to `maxFailedTasks` and `maxBlacklistedExecutors`, and change the comment to:

This strategy adds an executor to the blacklist for *all* tasks when the executor has too many task failures.  An executor is placed in the blacklist when there are more than [[maxFailedTasks]] failed tasks.  Furthermore, all executors in one node are put into the blacklist if there are more than [[maxBlacklistedExecutors]] blacklisted executors on one node.  The benefit of this strategy is that different taskSets can learn experience from other taskSet to avoid allocating tasks on problematic executors.